src/aio/aio_theme_switch.py

```diff
@@ -61,118 +59,60 @@ def __init__(
         - ThemeSwitchAIO.ids.store(aio_id)
         """
 
-        from dash_bootstrap_templates import load_figure_template
-
-        load_figure_template(dbc_themes_lowercase)
-
-        if aio_id is None:
-            aio_id = str(uuid.uuid4())
-
-        if themes is None:
-            themes = [dbc.themes.BOOTSTRAP, dbc.themes.CYBORG]
+        # init icons and switch_props
         if icons is None:
             icons = {"left": "fa fa-moon", "right": "fa fa-sun"}
-
-        switch_props = switch_props.copy()
-        if "value" not in switch_props:
-            switch_props["value"] = True
-        if "className" not in switch_props:
-            switch_props["className"] = "d-inline-block ms-1"
+        if switch_props is None:
+            switch_props = {}
+        # set "value" and "className" if they don't exist
+        switch_props.setdefault("value", True)
+        switch_props.setdefault("className", "d-inline-block ms-1")
+
+        # add fontawesome resource for the icons and dbc.css to apply Bootstrap theme to dcc, DataTable and Dash AG Grid
+        app = get_app()
+        app.config.external_stylesheets += [
```

Comment:
This is a much better solution than using a callback to add the stylesheet!

What do you think about bundling the css - at least the default icons for the light and dark? That way it would work even without an internet connection.  I found an example in this [forum post](
https://community.plotly.com/t/introducing-agridable-effortlessly-format-tabular-data-in-your-dash-app/84815/4?u=annmariew)

Do you think this will cause conflicts if someone is also using a has a different fontawesome version in their app? I know I was doing it this way in the callback, but it just occurred to me that it could be a problem.

Reply:
I tried to bundle fontawesome but that's not that easy, there are a bunch of files that have to be added alongside.
Those are .eot .svg .ttf .woff .woff2 files...
First time I see those types of file (except .svg 😄), I guess that for the images of the icons themselves. I don't know if we can only keep the default icons.
So I think it's better to keep the external link.

You're right, if the user add a different version, it's loaded before the version of the AIO...
But I used 
`app.config.external_stylesheets.insert(0, "https://use.fontawesome.com/releases/v5.15.4/css/all.css")`
to add it before the version of the user: problem solved! 🎉

Reply:
Good solution - thanks! :rocket: